WORKSPACE

```diff
@@ -46,3 +47,35 @@ load("@com_google_googleapis//:repository_rules.bzl", "switched_rules_by_languag
 switched_rules_by_language(
     name = "com_google_googleapis_imports",
 )
+
+http_archive(
+    name = "io_bazel_rules_go",
+    sha256 = "a82a352bffae6bee4e95f68a8d80a70e87f42c4741e6a448bec11998fcc82329",
+    urls = ["https://github.com/bazelbuild/rules_go/releases/download/0.18.5/rules_go-0.18.5.tar.gz"],
+)
+
+load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_dependencies")
+
+go_rules_dependencies()
+
+http_archive(
+    name = "com_github_grpc_grpc",
+    sha256 = "727fae92372f0b6e68f969e2b200add7a87e1395055a72d6363cc82942f93000",
+    strip_prefix = "grpc-1.30.0-pre1",
```

Comment:
Should we depend on a non-rc release?

Reply:
The main reason I went with this instead of 1.29 was because 1.29 is a bit old now, and 1.30 should be out soon. I can add a TODO to switch to a 1.30 stable, or go with 1.29. I don't think I mind too much. Do you have a preference?

Reply:
IIUC your proto did not add any dependency to grpc; the dependency comes transitively from Envoy. Is that Right? If so, it's probably best to use the same version as Envoy does.

Reply:
You are right about the dependency to gRPC from envoy. Currently envoy depends on gRPC v1.25. Is there a way to tell bazel to depend on the same version as envoy does instead of hardcoding that value here (otherwise how do we keep track of things when envoy changes to point to a different version)?

There is a way to pull in all dependencies of envoy, but this brings in too much for us: https://github.com/envoyproxy/envoy/blob/12a4bc430eaf440ceb0d11286cfbd4c16b79cdd1/bazel/repositories.bzl#L131

On a side note, should we also provide a function in some bazel file to pull in all our dependencies to make life easier for people who depend on grpc-proto?

Reply:
grpc-java uses https://github.com/grpc/grpc-java/blob/master/buildscripts/sync-protos.sh to import grpc-proto protos so the bazel enhancements won't work there. The plan is to add  grpc-xds to sync-protos.sh so it will add these protos to `xds/src/main/proto/...` . Because the xds project already has envoy protos (and specifically the one defining ApiConfigSource) I don't expect any problems for the grpc-xds project but could be a problem for the other projects (currently alts grpclb services rls). (FYI @ejona86)

Reply:
This does seem a little messy, because it basically adds a circular dependency.  C-core will not be able to depend on this if this depends on C-core.

That having been said, since we are currently in a world of copying these protos into the language-specific repos instead of depending on this repo as a bazel dependency and/or submodule, this is probably okay.  It's just unfortunate, because it adds yet another barrier to ever moving to a saner dependency model in the future.

I would suggest an alternative, but I don't see a better one. :(

Reply:
Done. Updated to the latest release.